core/src/main/scala/com/pingcap/tispark/TiDBRelation.scala

```diff
@@ -73,10 +72,16 @@ class TiDBRelation(session: TiSession, tableRef: TiTableReference, meta: MetaMan
       )
       .head
 
+    val downgradeDagRequest = dagRequest.copy()
+    // We need to clear index info in order to perform table scan
+    downgradeDagRequest.clearIndexInfo()
+    downgradeDagRequest.resetFilters(downgradeDagRequest.getDowngradeFilters)
```

Comment:
I think maintaining `downgradeDagRequest` out of `RegionTaskExec` increases coupling of downgrade logic and it's caller, it might be more OOP to make downgrade logic fully encapsulated in `RegionTaskExec` and transparent to it's caller.

Reply:
you are right.